source/extensions/filters/http/adaptive_concurrency/controller/gradient_controller.cc

```diff
@@ -109,13 +109,15 @@ void GradientController::enterMinRTTSamplingWindow() {
 }
 
 void GradientController::updateMinRTT() {
-  ASSERT(inMinRTTSamplingWindow());
-
   {
     absl::MutexLock ml(&sample_mutation_mtx_);
-    min_rtt_ = processLatencySamplesAndClear();
-    stats_.min_rtt_msecs_.set(
-        std::chrono::duration_cast<std::chrono::milliseconds>(min_rtt_).count());
+    if (hist_sample_count(latency_sample_hist_.get()) > 0) {
+      min_rtt_ = processLatencySamplesAndClear();
+      stats_.min_rtt_msecs_.set(
+          std::chrono::duration_cast<std::chrono::milliseconds>(min_rtt_).count());
+    } else {
+      stats_.empty_histogram_.inc();
```

Comment:
This ought to be an error log.

Reply:
Have updated the logic a little bit, and exit the method when `else` code path is hit. Just as mentioned in the [comment](https://github.com/envoyproxy/envoy/pull/15004#discussion_r578011563), this `else`  code path will always have the possibility to hit as long as updateMinRTT() get invoked simultaneously, in this case i am not sure whether we should log it as an error though.